File: lib/src/logs/logger.dart
import 'package:flutter/foundation.dart';

// ignore_for_file: avoid_print
class Logger {
  Logger._internal();

  static RegExp stackTraceRegex = RegExp(r'#2\s+.*:(\d+):\d+\)');
  static String _getLastLine() {
    String stack = StackTrace.current.toString();
    return stackTraceRegex.firstMatch(stack)?.group(1) ?? '?';
  }

  static void d(String className, String msg) {
    debugPrint(
        '[Awesome Notifications - DEBUG]: $msg ($className:${_getLastLine()})');
  }

  static void e(String className, String msg) {
    print(
        '\x1B[31m[Awesome Notifications - ERROR]: $msg ($className:${_getLastLine()})\x1B[0m');
  }

  static void i(String className, String msg) {
    print('\x1B[34m[Awesome Notifications - INFO]: $msg ($className)\x1B[0m');
  }

  static void w(String className, String msg) {
    print(
        '\x1B[33m[Awesome Notifications - WARNING]: $msg ($className:${_getLastLine()})\x1B[0m');
  }
}

File: ios/Classes/lib/awesome_notifications_core/logs/Logger.swift
//
//  Logger.swift
//  awesome_notifications
//
//  Created by CardaDev on 01/04/22.
//

import Foundation
import os.log

public class Logger {
    
    static func d(_ className:String, _ message:String, line: Int = #line){
        os_log("Swift: [AWESOME NOTIFICATIONS - DEBUG] %@ (%@:%d)", type: .debug, message, className, line)
    }
    
    static func e(_ className:String, _ message:String, line: Int = #line){
        os_log("Swift: [AWESOME NOTIFICATIONS - ERROR] %@ (%@:%d)", type: .error, message, className, line)
    }
    
    static func i(_ className:String, _ message:String, line: Int = #line){
        os_log("Swift: [AWESOME NOTIFICATIONS - INFO] %@ (%@:%d)", type: .info, message, className, line)
    }
    
    static func w(_ className:String, _ message:String, line: Int = #line){
        os_log("Swift: [AWESOME NOTIFICATIONS - WARNING] %@ (%@:%d)", type: .fault, message, className, line)
    }
}
